editor/src/app/sites/sites-state/sites.state.ts

```diff
@@ -37,94 +45,120 @@ export class SitesState implements NgxsOnInit {
 
 
   @Action(CreateSiteAction)
-  createSite({ setState, getState }: StateContext<SiteStateModel[]>) {
-    const currentState = getState();
-    const newSite: SiteStateModel = {
-      // @todo sync with backend
-      // @todo get unique name from backend
-      name: 'untitled-' + Math.random().toString(36).substr(2, 9),
-      title: '',
-      order: currentState.length,
-      '@attributes': {
-        published: 0
-      }
-    };
-
-    setState(
-      [...currentState, newSite]
-    );
+  createSite({ setState, getState }: StateContext<SiteStateModel[]>, action: CreateSiteAction) {
+    const siteName = action.site ? (action.site.name === '' ? '0' : action.site.name) : '-1';
+    this.appStateService.sync('sites', { site: siteName }, 'POST')
+      .then((response: any) => {
```

Comment:
Let's keep to observables since we're already using them everywhere. So return observable from `sync` (remove the `toPromise`). And add `.pipe(map(response => {...}))` here.